source/camera/CullingManager.cpp

```diff
@@ -0,0 +1,253 @@
+    // create a slope -- a set of left and rigth path
+    mWalk.convex_hull();
+    mWalk.buildSlopes();
+
+    OD_LOG_DBG( mOldWalk.debug());
+    OD_LOG_DBG( mWalk.debug());        
+
+    // reset index pointers to the begging of collections
+    mOldWalk.prepareWalk();
+    mWalk.prepareWalk();
+
+    newBashAndSplashTiles(HIDE);
+    showAllTiles();
+
+}
+
+void CullingManager::hideAllTiles(void)
+{
+    GameMap* gm = mCm->getGameMap();
+    if (!gm)
+        return;
+
+    for (int jj = 0; jj < gm->getMapSizeY() ; ++jj)
+    {
+        for (int ii = 0; ii < gm->getMapSizeX(); ++ii)
+        {
+            RenderManager::getSingleton().rrDetachEntity(gm->getTile(ii, jj));
+        }
+    }
+}
+
+void CullingManager::showAllTiles(void)
+{
+    GameMap* gm = mCm->getGameMap();
+    if (!gm)
+        return;
+
+    for (int jj = 0; jj < gm->getMapSizeY() ; ++jj)
+    {
+        for (int ii = 0; ii < gm->getMapSizeX(); ++ii)
+        {
+            RenderManager::getSingleton().rrAttachEntity(gm->getTile(ii, jj));
+        }
+    }
+}
+
+void CullingManager::newBashAndSplashTiles(int64_t mode){
+    int64_t xxLeftOld = mOldWalk.getTopLeftVertex().x;
+    int64_t xxRightOld= mOldWalk.getTopRightVertex().x;
+    int64_t xxLeft = mWalk.getTopLeftVertex().x;
+    int64_t xxRight= mWalk.getTopRightVertex().x;
+    int64_t xxp, yyp;
+    std::stringstream ss;
+    int64_t bb = (( std::min(mWalk.getBottomLeftVertex().y , mOldWalk.getBottomRightVertex().y) >> PRECISION_DIGITS) - 2) << PRECISION_DIGITS;
+
+    for (int64_t yy = ((std::max(mWalk.getTopLeftVertex().y , mOldWalk.getTopRightVertex().y  ) >> PRECISION_DIGITS) + 2) << PRECISION_DIGITS;  yy >= bb; yy -= UNIT)
+    {
+        mOldWalk.notifyOnMoveDown(yy);
+        mWalk.notifyOnMoveDown(yy);
+        xxLeft = mWalk.getCurrentXLeft(yy);
+        xxLeftOld = mOldWalk.getCurrentXLeft(yy);
+        xxRight = mWalk.getCurrentXRight(yy);
+        xxRightOld = mOldWalk.getCurrentXRight(yy);
+   
+        int64_t mm = ((std::min(xxLeft, xxLeftOld) >> PRECISION_DIGITS) << PRECISION_DIGITS) ;
+        if(std::min(xxLeft, xxLeftOld) < std::max(xxRight,xxRightOld) )
+        {
+            for (int64_t xx = mm ; xx <= std::max(xxRight,xxRightOld) ; xx+= UNIT)
+            {
+                bool bash = (xx >= xxLeftOld && xx <= xxRightOld && (yy >= mOldWalk.getBottomLeftVertex().y) && yy <= mOldWalk.getTopLeftVertex().y);
+                bool splash = (xx >= xxLeft && xx <= xxRight && (yy >= mWalk.getBottomLeftVertex().y) && yy <= mWalk.getTopLeftVertex().y);
+
+                xxp = ((xx >>PRECISION_DIGITS));
+                yyp = ((yy >>PRECISION_DIGITS));
+                GameMap* gm = mCm->getGameMap();
+                if(bash && splash && (mode & HIDE) && (mode & SHOW))
+                {
+                    // Nothing
+                }
+
+                else if (gm && bash && (mode & HIDE) && xxp>=0 && yyp>= 0 && xxp<mCm->getGameMap()->getMapSizeX() && yyp < mCm->getGameMap()->getMapSizeY())
```

Comment:
Well, I have nothing against it but I would prefer to have real arguments instead of "using break is the first step to hell".
But since you don't feel like giving ones, I will give mines for using break/continue.
1 - It allows to clearly separate different parts of code that should stop or go to next iteration. In my code, it is pretty obvious that "if(bash && splash && (mode & HIDE) && (mode & SHOW))", we don't want to proceed the tile. In yours, you have to check that all other conditions are "else if" and not if of anything else.
2 - It allows to isolate conditions that should stop processing the loop. For example, if we assume you want to proceed tiles with X even, Y even, fullness > 0 and type = dirt, your code would be something like:
```
for(Tile* tile : tiles)
{
   if((tile->getX() % 2) || (tile->getY() % 2) || (tile->getFullness() <= 0) || (tile->getType() != TileType::Dirt))
   {
      // Nothing
   }
   else
   {
      // Processing code
   }
}
```
If, for some reason, you or someone else have to re-write this code, he is likely to forget a condition and at review, it is easy to miss the forgotten part. And I am talking about a real case since this exact problem happened to me when improving the room absorbtion code. It have been merged without seeing I had forgotten one of the conditions. Thus, we had a pretty nasty segfault in some specific cases.

My code would be more like:
```
for(Tile* tile : tiles)
{
   if(tile->getX() % 2)
      continue;
   if(tile->getY() % 2)
      continue;
   if(tile->getFullness() <= 0)
      continue;
   if(tile->getType() != TileType::Dirt)
      continue;

   // Processing code
```
I guess you will agree that it is not as likely to miss a conditions like that.

3 - It allows to have less imbricated code. See my previous example.

4 - Because the conditions are more isolated, it allows to see what has been changed at the first glance at code review. For example, if you add or remove one of the conditions in the previous example, in your code, that will change the one line with the list of conditions. In my version, you will add/remove a specific part of the code that will be easy to identify.
5 - For the same reason as 4, it helps keeping code history more readable.

To be honest, before I joined OD team I had the same feeling as yours about continue/break and I never used it (I had never thought about it but since my teachers always told me that using them was bad, I didn't use them). But after seeing it used in OD code and thinking about it, it became obvious that it is much better as the only reason I see to not use them is the same you gave so far: "it is bad to use them". But when I think about it, all the reasons I can find tell me the opposite. And keeping code history clean and better readability is something that should be above everything else, especially in an open source project like this one.